contracts/marketpools.js

```diff
@@ -0,0 +1,350 @@
+      await api.db.insert('liquidityPositions', newlp);
+    }
+
+    // deposit requested tokens to contract
+    const baseRes = await api.executeSmartContract('tokens', 'transferToContract', { symbol: baseSymbol, quantity: baseQuantity, to: 'marketpools' });
+    const quoteRes = await api.executeSmartContract('tokens', 'transferToContract', { symbol: quoteSymbol, quantity: quoteQuantity, to: 'marketpools' });
+    if (!api.assert(baseRes.errors === undefined && quoteRes.errors === undefined, 'deposit transfer errors')) return;
+    await updatePoolStats(pool, baseQuantity, quoteQuantity, false);
+    api.emit('addLiquidity', { baseSymbol, quoteSymbol });
+  }
+};
+
+actions.removeLiquidity = async (payload) => {
+  const {
+    tokenPair,
+    baseQuantity,
+    quoteQuantity,
+    isSignedWithActiveKey,
+  } = payload;
+
+  if (!api.assert(isSignedWithActiveKey === true, 'you must use a transaction signed with your active key')
+    || !api.assert(baseQuantity && api.BigNumber(baseQuantity).gt(0), 'invalid baseQuantity')
+    || !api.assert(quoteQuantity && api.BigNumber(quoteQuantity).gt(0), 'invalid quoteQuantity')
+    || !await validateTokenPair(tokenPair)) {
+    return;
+  }
+
+  const [baseSymbol, quoteSymbol] = tokenPair.split(':');
+  const baseToken = await api.db.findOneInTable('tokens', 'tokens', { symbol: baseSymbol });
+  const quoteToken = await api.db.findOneInTable('tokens', 'tokens', { symbol: quoteSymbol });
+  if (!api.assert(api.BigNumber(baseQuantity).dp() <= baseToken.precision, 'baseQuantity precision mismatch')
+    || !api.assert(api.BigNumber(quoteQuantity).dp() <= quoteToken.precision, 'quoteQuantity precision mismatch')) return;
+
+  const pool = await api.db.findOne('pools', { tokenPair });
+  if (api.assert(pool, 'no existing pool for tokenPair')) {
+    if (!api.assert(api.BigNumber(pool.baseQuantity).gt(0)
+      && api.BigNumber(pool.quoteQuantity).gt(0), 'insufficient liquidity')
+      || !validateLiquiditySwap(pool, baseQuantity, quoteQuantity)) return;
+
+    const lp = await api.db.findOne('liquidityPositions', { account: api.sender, tokenPair });
+    if (api.assert(lp, 'no existing liquidity position')
+      && api.assert(api.BigNumber(lp.baseQuantity).minus(baseQuantity).gte(0)
+      && api.BigNumber(lp.quoteQuantity).minus(quoteQuantity).gte(0), 'not enough liquidity in position')) {
+      lp.baseQuantity = api.BigNumber(lp.baseQuantity).minus(baseQuantity);
+      lp.quoteQuantity = api.BigNumber(lp.quoteQuantity).minus(quoteQuantity);
+      if (api.BigNumber(lp.baseQuantity).eq(0) && api.BigNumber(lp.quoteQuantity).eq(0)) {
+        await api.db.remove('liquidityPositions', lp);
+      } else {
+        await api.db.update('liquidityPositions', lp);
+      }
+
+      await api.transferTokens(api.sender, baseSymbol, baseQuantity, 'user');
+      await api.transferTokens(api.sender, quoteSymbol, quoteQuantity, 'user');
+      await updatePoolStats(pool, -baseQuantity, -quoteQuantity, false);
```

Comment:
Change unary minus to BigNumber . negated()

Reply:
Changed